Code/Framework/AzCore/AzCore/Task/TaskExecutor.cpp

```diff
@@ -308,11 +308,11 @@ namespace AZ
 
     void TaskExecutor::SetInstance(TaskExecutor* executor)
     {
-        if (!executor)
+        if (!executor) // allow unsetting the executor
         {
             s_executor.Reset();
         }
-        else if (!s_executor) // ignore any calls to set after the first (this happens in unit tests that create new system entities)
+        else if (!s_executor) // ignore any extra executors after the first (this happens during unit tests)
```

Comment:
I'd say at least do a warning here because a unit test may results from a specific instance, but if its ignored without any indication, it will be difficult to troubleshoot any unexpected behavior